docs/01_nodeos/02_usage/03_development-environment/10_local-single-node-testnet-consensus.md

```diff
@@ -0,0 +1,199 @@
+```
+
+[[warning | Security Notice]]
+| Do not use the parameters `--access-control-allow-origin='*'`, `--http-validate-host=false`, `--verbose-http-errors` and `--contracts-console`, in production because they either weaken the security or affect performance of your node.
+
+After running `nodeos`, you should get log messages similar as below. It means the blocks are successfully produced.
+
+```console
+info  2021-01-07T15:59:07.902 thread-0  producer_plugin.cpp:2053      produce_block        ] Produced block 98fa5cdd7ce06ae8... #162 @ 2021-01-07T15:59:08.000 signed by eosio [trxs: 0, lib: 161, confirmed: 0]
+info  2021-01-07T15:59:08.401 thread-0  producer_plugin.cpp:2053      produce_block        ] Produced block 972190051a840992... #163 @ 2021-01-07T15:59:08.500 signed by eosio [trxs: 0, lib: 162, confirmed: 0]
+info  2021-01-07T15:59:08.901 thread-0  producer_plugin.cpp:2053      produce_block        ] Produced block d8727439a26f36f6... #164 @ 2021-01-07T15:59:09.000 signed by eosio [trxs: 0, lib: 163, confirmed: 0]
+info  2021-01-07T15:59:09.401 thread-0  producer_plugin.cpp:2053      produce_block        ] Produced block 6f2302aeb944c5ca... #165 @ 2021-01-07T15:59:09.500 signed by eosio [trxs: 0, lib: 164, confirmed: 0]
+...
+info  2021-01-07T15:59:10.902 thread-0  producer_plugin.cpp:2053      produce_block        ] Produced block 8cd2384a28818e19... #168 @ 2021-01-07T15:59:11.000 signed by eosio [trxs: 0, lib: 167, confirmed: 0]
+...
+info  2021-01-07T15:59:11.900 thread-0  producer_plugin.cpp:2053      produce_block        ] Produced block c8c82b50249e9f6d... #170 @ 2021-01-07T15:59:12.000 signed by eosio [trxs: 0, lib: 169, confirmed: 0]
+...
+```
+
+At this point, `nodeos` is running with a single producer, `eosio`.
+
+### 3. Preactivate Protocol Features
+
+All of the protocol upgrade features introduced in v1.8 and on subsequent versions require a special protocol feature (also known as `PREACTIVATE_FEATURE`) to be activated and for an updated version of the system contract that makes use of the functionality introduced by that feature to be deployed.
+
+To activate the special protocol `PREACTIVATE_FEATURE` run the following command:
+
+```sh
+curl --request POST \
+    --url http://127.0.0.1:8888/v1/producer/schedule_protocol_feature_activations \
+    -d '{"protocol_features_to_activate": ["0ec7e080177b2c02b278d5088611686b49d739925a92d9bfcacd7fc6b74053bd"]}'
+```
+
+### 4. Get The System Smart Contracts
+
+Two reference system smart contracts, the `eosio.boot` and `eosio.bios`, are located in the [`eos`](https://github.com/EOSIO/eos.git) repository, with source code and their binary form. You can build them from source or use them directly.
+
+#### 4.1 Use The Prebuilt System Smart Contracts
+
+To use the prebuilt system smart contract execute the following command lines:
+
+```sh
+cd ~
+git clone https://github.com/EOSIO/eos.git
+cd ./eos/contracts/contracts/contracts/
```

Comment:
needs fix (you have an extra `contracts`)

Reply:
tha path is correct, it has `contracts` twice in it.

Reply:
Still needs fixing. Since it's the third time I raise the mistake, let me count the number of `contracts` instances in your `cd` command for you:
```
cd ./eos/contracts/contracts/contracts/
             ^         ^         ^
            one       two      three
```
As I said in my first post, the correct command should read:
```
cd ./eos/contracts/contracts/
```

Reply:
aaaarrhhhh one more left behind!